mod/settings.php

```diff
@@ -308,7 +308,13 @@ function settings_post(App $a)
 
 	if ($email != $user['email']) {
 		//  check for the correct password
-		if (!User::authenticate(intval(local_user()), $_POST['mpassword'])) {
+		try {
+			$authenticated = User::getIdFromPasswordAuthentication(intval(local_user()), $_POST['mpassword']);
+		} catch (Exception $ex) {
+			$authenticated = false;
+		}
+
+		if (!$authenticated) {
```

Comment:
No need for a variable if you're just checking credentials.
```suggestion
		try {
			User::getIdFromPasswordAuthentication(local_user(), $_POST['mpassword']);
		} catch (Exception $ex) {
```